test_util/testharness.h

```diff
@@ -15,13 +15,39 @@
 #include <gtest/gtest.h>
 #endif
 
-// If GTEST_SKIP is available, use it. Otherwise, define skip as success
+// If GTEST_SKIP is available, use it. Otherwise, define skip as success.
+//
+// A "skipped" test has a specific meaning in Facebook infrastructure: a
+// precondition was not met, and there is a testing hole if any
+// parameterization of a test (e.g. Foo/FooTest.Bar/42) is skipped for all
+// tested build configurations/platforms/etc.
```

Comment:
That should be skipped. There is a testing hole if the test is never run.

Reply:
I am still not sure I understand when to use SKIP versus BYPASS.  More concrete examples would help.  If functionality is not supported on a given platform (Windows, Mac, etc), which do I use?  If functionality is not compiled in (Snappy, JeMalloc), which do I use?  What if something is not supported by the specific environment or configuration?

Right now, most of the tests simply ignore those conditions and make the tests look like they pass and did something.  I fully support this effort as it would be nice to know these tests that were not fully executed, but think the rules on what "SKIP" versus "BYPASS" means should be called out in greater detail.